components/director/internal/domain/formation/notifications.go

```diff
@@ -778,3 +799,29 @@ func (ns *notificationsService) createWebhookRequest(ctx context.Context, webhoo
 		CorrelationID: correlation.CorrelationIDFromContext(ctx),
 	}, nil
 }
+
+func (ns *notificationsService) extractCustomerTenantContext(ctx context.Context, internalTenantID string) (*webhookdir.CustomerTenantContext, error) {
```

Comment:
i am not very familiar with the structure of the code about the formations, but i noticed that this function has a "copy" of itself in the `/formationassignment/notifications.go`. Is it possible to define the function once and somehow both files use it?

Reply:
The main thing is that this function has two different receivers in both cases (`notificationsService` and `formationAssignmentNotificationService`) that use their own repository layer functions.

Currently, the used repo layer functions coincide but we don't have any guarantee that in the future they will always be the same/with the same signature depending on the flow. That's why I separated them.